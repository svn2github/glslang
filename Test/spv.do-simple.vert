#version 300 es
void main() {
  int i = 0;
  do {
    i++;
  } while(i<10);
}
