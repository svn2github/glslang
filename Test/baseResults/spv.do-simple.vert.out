spv.do-simple.vert



Linked vertex stage:



// Module Version 99
// Generated by (magic number): 51a00bb
// Id's are bound by 26

                              Source GLSL 300
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4
                              Name 4  "main"
                              Name 9  "i"
                              Name 24  "gl_VertexID"
                              Name 25  "gl_InstanceID"
                              Decorate 9(i) PrecisionHigh 
                              Decorate 24(gl_VertexID) PrecisionHigh 
                              Decorate 24(gl_VertexID) BuiltIn VertexId
                              Decorate 24(gl_VertexID) NoStaticUse 
                              Decorate 25(gl_InstanceID) PrecisionHigh 
                              Decorate 25(gl_InstanceID) BuiltIn InstanceId
                              Decorate 25(gl_InstanceID) NoStaticUse 
               2:             TypeVoid
               3:             TypeFunction 2 
               7:             TypeInt 32 1
               8:             TypePointer Function 7(int)
              10:      7(int) Constant 0
              16:      7(int) Constant 1
              19:      7(int) Constant 10
              20:             TypeBool
              23:             TypePointer Input 7(int)
 24(gl_VertexID):     23(ptr) Variable Input 
25(gl_InstanceID):     23(ptr) Variable Input 
         4(main):           2 Function None 3
               5:             Label
            9(i):      8(ptr) Variable Function 
                              Store 9(i) 10 
                              Branch 11
              11:             Label
                              LoopMerge 12 None
                              Branch 13
              13:             Label
              15:      7(int) Load 9(i) 
              17:      7(int) IAdd 15 16
                              Store 9(i) 17 
                              Branch 14
              14:             Label
              18:      7(int) Load 9(i) 
              21:    20(bool) SLessThan 18 19
                              BranchConditional 21 22 12 
              22:               Label
                                Branch 11
              12:             Label
                              Branch 6
               6:             Label
                              Return
                              FunctionEnd
